pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-standalone/src/main/java/org/apache/pinot/plugin/ingestion/batch/standalone/SegmentGenerationJobRunner.java

```diff
@@ -253,6 +264,15 @@ private void submitSegmentGenTask(File localTempDir, URI inputFileURI, int seqId
     taskSpec.setFailOnEmptySegment(_spec.isFailOnEmptySegment());
     taskSpec.setCustomProperty(BatchConfigProperties.INPUT_DATA_FILE_URI_KEY, inputFileURI.toString());
 
+    // If there's already been a failure, log and skip this file. Do this check right before the
+    // submit to reduce odds of starting a new segment when a failure is recorded right before the
+    // submit.
+    if (_failure.get() != null) {
+      LOGGER.info("Skipping Segment Generation Task for {} due to previous failures", inputFileURI);
+      _segmentCreationTaskCountDownLatch.countDown();
```

Comment:
(minor) This count down is not required because the previous failure should already drain it

Reply:
Thanks, removed.